external/src/main/java/com/redhat/cloud/policies/engine/process/Receiver.java

```diff
@@ -184,15 +184,19 @@
             List<Event> eventList = new ArrayList<>(1);
             eventList.add(event);
             if (storeEvents) {
-                alertsService.addEvents(eventList);
+                return alertsService.addEvents(eventList)
+                        .replaceWith(ack(input));
             } else {
-                alertsService.sendEvents(eventList);
+                return alertsService.sendEvents(eventList)
+                        .replaceWith(ack(input));
```

Comment:
This is the place where the processing was changed from blocking to non-blocking.

Reply:
If there is an error in the `sendEvents`, it won't `ack` the input?
or it `ack`s it regardless of the outcome?

Reply:
An exception will be thrown and `ack` will never be called. It should behave like the old code. Maybe the exception thrown will be different though, Mutiny may wrap the original exception into a Mutiny exception. But I don't think it will change anything in terms of application behavior.